aim/sdk/adapters/keras_tuner.py

```diff
@@ -0,0 +1,66 @@
+from typing import Optional, List
+
+from aim.sdk.run import Run
+from aim.ext.resource.configs import DEFAULT_SYSTEM_TRACKING_INT
+
+try:
+    from kerastuner.engine.tuner_utils import TunerCallback
+except ImportError:
+    raise RuntimeError(
+        'This contrib module requires KerasTuner Lightning to be installed. '
+        'Please install it with command: \n pip install keras-tuner'
+    )
+
+
+class AimCallback(TunerCallback):
+    def __init__(self, tuner=None,
+                 repo: Optional[str] = None,
+                 experiment: Optional[str] = None,
+                 system_tracking_interval: int = DEFAULT_SYSTEM_TRACKING_INT,
+                 log_system_params: bool = True):
+        self.tuner = tuner
+        self._repo_path = repo
+        self._experiment_name = experiment
+        self._system_tracking_interval = system_tracking_interval
+        self._log_system_params = log_system_params
+
+        self._started_trials: List[str] = []
+        self.trial = None
+        self._run = None
+
+    @property
+    def experiment(self) -> Run:
+        if self._run is not None:
+            return self._run
+
+    def on_epoch_begin(self, epoch, logs=None):
```

Comment:
What about creating a separate run for each trial and group them by an experiment? So:
- aim.Experiment - keras-tuner run
- aim.Run - a single trial, loging trial metrics and hparams

Reply:
@gorarakelyan that's exactly how we do.

Reply:
ah, I was confused seeing the trial_id context key on metrics. Then probably it is worth removing it from the context and attaching to run? run['trial_id'] = TRIAL_ID

the rest looks great!!

Reply:
Sure, totally makes sense.